tests/js/pieTests.js

```diff
@@ -0,0 +1,182 @@
+/*!
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.registerNamespace("gpii.tests.chartAuthoring");
+
+    fluid.defaults("gpii.tests.chartAuthoring.pieChart.pie", {
+        gradeNames: ["gpii.chartAuthoring.pieChart.pie", "autoInit"],
+        pieOptions: {
+            width: 200,
+            height: 200,
+            colors: ["#000000", "#ff0000", "#00ff00", "#0000ff", "#aabbcc", "#ccbbaa"]
+        },
+        listeners: {
+            "onPieCreated.addMouseoverListener": {
+                listener: "{that}.addD3Listeners",
+                args: ["{that}.dom.slice", "mouseover", "gpii.tests.chartAuthoring.mouseOverListener"]
+            }
+        },
+        members: {
+            mouseOverListenerCalled: false
+        }
+    });
+
+    gpii.tests.chartAuthoring.mouseOverListener = function (data, i, that) {
+        that.mouseOverListenerCalled = true;
+    };
+
+    gpii.tests.chartAuthoring.validatePie = function (that, testSliceDataFunc) {
+        var pie = that.locate("pie");
+
+        // Test the drawing
+        jqUnit.assertNotEquals("The SVG element is created with the proper selector", 0, pie.length);
+        jqUnit.assertEquals("The width is set correctly on the pie chart", that.options.pieOptions.width, pie.attr("width"));
+        jqUnit.assertEquals("The height is set correctly on the pie chart", that.options.pieOptions.height, pie.attr("height"));
+        jqUnit.assertEquals("The pie slices have been created with the proper selectors", that.model.dataSet.length, that.locate("slice").length);
+        jqUnit.assertEquals("The texts for pie slices have been created with the proper selectors", that.model.dataSet.length, that.locate("text").length);
+
+        // Test that displayed values are in sync with the current model
+        gpii.tests.chartAuthoring.testPieTextSyncWithModelDataSet(that);
+
+        // Each slice receives the corresponding data object
+        var d3Slices = that.jQueryToD3($(that.locate("slice")));
+        d3Slices.each(function (d, i) {
+            jqUnit.assertEquals("The slice colors are filled correctly", that.options.pieOptions.colors[i], ($(this).attr("fill")));
```

Comment:
I don't find a good alternative to replace the use of "this". This is the way to access html elements when iterating over a d3 path elements array.